addons/web/static/src/views/calendar/calendar_common/calendar_common_renderer.js

```diff
@@ -137,17 +137,20 @@ export class CalendarCommonRenderer extends Component {
         return Object.values(this.props.model.records).map((r) => this.convertRecordToEvent(r));
     }
     convertRecordToEvent(record) {
-        const allDay = record.isAllDay || record.end.diff(record.start, "hours").hours >= 24;
+        const forceAllDay =
+            this.props.model.hasAllDaySlot && record.end.diff(record.start, "hours").hours >= 24;
+        const allDay = record.isAllDay || forceAllDay;
```

Comment:
Why?
`record.isAllDay` and `model.hasAllDaySlot` do the same thing: `fieldMapping.all_day || fields[fieldMapping.start_date].type === "date"`
Also, if this is really needed, this is a fix in the framework so it needs a test.

Reply:
You're right. I did not realize that `hasAllDaySlot` would be truthy if we had an all_day attribute set on the arch. Thanks for the review. I removed all changes to web and it seems to be working.